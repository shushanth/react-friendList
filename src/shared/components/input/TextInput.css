:local(.textInput) {
  border-radius: 0;
  border-color: #abaaaa;
  border-left: 0;
  border-right: 0;
}
