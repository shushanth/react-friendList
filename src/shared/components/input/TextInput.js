/**
 * Generic component for input types (number, radio , text..),
 */
import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

import styles from "./TextInput.css";

const { textInput } = styles;

const TextInput = ({
  name,
  type,
  value = "",
  placeholder,
  focusAuto,
  onInputChange
}) => {
  return (
    <input
      name={name}
      className={classnames("form-control", textInput)}
      value={value}
      type={type}
      autoFocus={focusAuto}
      placeholder={placeholder}
      onChange={onInputChange}
      onKeyDown={onInputChange}
    />
  );
};

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  focusAuto: PropTypes.bool,
  onInputChange: PropTypes.func
};

export default TextInput;
