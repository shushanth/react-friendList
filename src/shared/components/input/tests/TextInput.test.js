import React from "react";
import { shallow } from "enzyme";

import TextInput from "../TextInput";
import styles from "../TextInput.css";

describe("Component: <TextInput/>", () => {
  let textInputWrapper;
  let props;
  beforeAll(() => {
    props = {
      name: "Test",
      type: "text",
      value: "",
      placeholder: "test placeholder",
      focusAuto: true,
      onInputChange: jest.fn()
    };
    textInputWrapper = shallow(<TextInput {...props} />);
  });

  it("should render without crashing", () => {
    expect(textInputWrapper.exists()).toBe(true);
    expect(textInputWrapper).toMatchSnapshot();
  });

  it("should have render input with attributes, matches props value", () => {
    let inputElement = textInputWrapper.find("input");
    expect(inputElement.props().name).toEqual(props.name);
    expect(inputElement.props().value).toEqual(props.value);
    expect(inputElement.props().autoFocus).toEqual(props.focusAuto);
    expect(inputElement.props().placeholder).toEqual(props.placeholder);
    expect(inputElement.props().type).toEqual(props.type);
  });

  describe("method: onInputChange", () => {
    it("should invoke onInputChange prop method, on input change", () => {
      let inputElement = textInputWrapper.find("input");
      inputElement.simulate("change");
      expect(props.onInputChange).toHaveBeenCalled();
    });

    it("should invoke onInputChange prop method, on input keydown", () => {
      let inputElement = textInputWrapper.find("input");
      inputElement.simulate("keyDown");
      expect(props.onInputChange).toHaveBeenCalled();
    });
  });
});
