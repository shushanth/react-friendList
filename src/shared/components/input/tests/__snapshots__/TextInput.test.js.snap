exports[`Component: <TextInput/> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    focusAuto={true}
    name="Test"
    onInputChange={[Function]}
    placeholder="test placeholder"
    type="text"
    value="" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "autoFocus": true,
      "className": "form-control",
      "name": "Test",
      "onChange": [Function],
      "onKeyDown": [Function],
      "placeholder": "test placeholder",
      "type": "text",
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "autoFocus": true,
        "className": "form-control",
        "name": "Test",
        "onChange": [Function],
        "onKeyDown": [Function],
        "placeholder": "test placeholder",
        "type": "text",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
